components/Header/NavLinks.tsx

```diff
@@ -0,0 +1,60 @@
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import React from 'react'
+
+interface NavLinksProps {
+    links: Array<NavLink>,
+    listStyle: string
+}
+
+type NavLink = {
+    href: string,
+    text: string
+}
+
+const NavLinks = ({ links, listStyle }: NavLinksProps) => {
+    const router = useRouter()
+    // const MyLink = React.forwardRef((props, ref) => <Link passHref={ref} {...props}/> );
+
+    return (
+        <ul className={listStyle}>
+            {
+                links.map((link, idx) => {
+                    /** Warns about passing a prop to href. 
+                     * Might need to undo functional approach here an manually enumerate each link.
+                     * */
+                    return (
+                        <li
+                        key={idx}
+                        className={`${router.pathname.split("?")[0] == link.href ? 'text-primary font-bold underline underline-offset-8' : 'text-neutral-700 dark:text-neutralDark-150'} hover:font-bold`}>
+                            <Link href={link.href} passHref> 
+                                <a>{link.text}</a>
+                            </Link>
+                        </li>
+                    )
+                })
+            }
+        </ul>
+    )
+}
+
+NavLinks.defaultProps = {
+    links: [
+        // TODO: Add as pages created
+        {
+            href: "/#", // change to /Search when search page created
+            text: "Search Page"
+        },
+        {
+            href: "/social",
+            text: "Social Feed"
```

Comment:
We may want to pad left-right these tab text and use "Search" "Feed" instead of "Search Page" and "Social Feed".
I also noticed that the hover effect pushing apart the other strings. One way to avoid that is to add a transparent padding by default.

Reply:
Let me know if the above makes sense.

Reply:
Fixed.